### PR TITLE
Spliting up bolt files for better readability

### PR DESCRIPTION
**Pull request by sketchthat**
Bolt files can get out of hand really quickly - I've found it better to break it up into separate files such as functions, rules and types.

Feel free to merge - comment or close if you don't want to go down this avenue.

I haven't moved all the types out into the `type` file - because I'm not sure the teams thoughts on it.

Also I've used `npm build` in the `package.json` file - I don't use yarn but I assume they are inter-changeable?

### CODE REVIEW

**Comment by tshedor** [comment]
@sketchthat Wow, had no idea you could use multiple bolt files. This is magic.

* Would you please replace the `npm run` with `yarn` in the package.json scripts? They have a similar syntax, but I'd rather stay with one package manager (npm also generates a different lockfile)
* Please add `bolt` to `build` in scripts (`... dist && yarn bolt`)
* Please run `yarn install` since you added a new devDep and the lockfile needs to be aware of it

File: bolt/database.rules.bolt [hunk 1]
```diff
@@ -0,0 +1,113 @@
+type Event {
```
Comment (sketchthat) [inline bolt/database.rules.bolt]: Personally I would move these into the `database.types.bolt` but its up to the rest of the team - thoughts?

Comment (tshedor) [reply bolt/database.rules.bolt]: Go for it

**Comment by tshedor** [comment]
Beauty, thanks @sketchthat

**Pull request merged**